validation-test/stdlib/HashedCollectionFilter3.swift

```diff
@@ -1,4 +1,4 @@
-// RUN: %target-run-stdlib-swift-swift3
+// RUN: %target-run-stdlib-swift
```

Comment:
Can this file be removed?

The `Sequence.filter(_:)` methods are also tested in `validation-test/stdlib/HashedCollectionFilter4.swift` (which currently `xfail`s its Swift 4 tests).